### ISSUE TITLE
Kube-mgmt can't delete ingresses it created in OPA

### ISSUE THREAD

**Issue by rlguarino** [open-policy-agent/kube-mgmt#7]
My kube-mgmt instance can't delete the ingresses that it added to the data section of the opa sidecar.

From kube-mgmt: 
```
time="2018-02-06T17:13:52Z" level=error msg="Failed to remove u-nicholas/examplesvc-ingress-debug" 
time="2018-02-06T17:13:52Z" level=error msg="Failed to remove u-nicholas/examplesvc-ingress" 
```

From opa:
```
sses/u-nicholas/examplesvc-ingress
time="2018-02-06T17:13:52Z" level=info msg="Sent response." client_addr="127.0.0.1:43816" req_id=545957 req_method=PUT req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress resp_bytes=0 resp_duration=0.219423 resp_status=204
time="2018-02-06T17:13:52Z" level=info msg="Received request." client_addr="127.0.0.1:43816" req_id=545958 req_method=PUT req_params="map[]" req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress-debug
time="2018-02-06T17:13:52Z" level=info msg="Sent response." client_addr="127.0.0.1:43816" req_id=545958 req_method=PUT req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress-debug resp_bytes=0 resp_duration=0.112821 resp_status=204
time="2018-02-06T17:13:52Z" level=info msg="Received request." client_addr="127.0.0.1:43816" req_id=545959 req_method=PUT req_params="map[]" req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress
time="2018-02-06T17:13:52Z" level=info msg="Sent response." client_addr="127.0.0.1:43816" req_id=545959 req_method=PUT req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress resp_bytes=0 resp_duration=0.129813 resp_status=204
time="2018-02-06T17:13:52Z" level=info msg="Received request." client_addr="127.0.0.1:43816" req_id=545960 req_method=PUT req_params="map[]" req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress-debug
time="2018-02-06T17:13:52Z" level=info msg="Sent response." client_addr="127.0.0.1:43816" req_id=545960 req_method=PUT req_path=/v1/data/kubernetes/ingresses/u-nicholas/examplesvc-ingress-debug resp_bytes=0 resp_duration=0.102676 resp_status=204
```

I'm not sure if I'm reading the Rest API docs right, but there's not much about removing existing data.

The ingress is still there if I curl the opa API directly.

**Comment by tsandall** [comment]
The [Data API](http://www.openpolicyagent.org/docs/rest-api.html#data-api) supports a PATCH operation that kube-mgmt should be using to remove resources.

Could you try upgrading to v0.5 of kube-mgmt? It includes a change that puts the error reason into the log message. From looking a thte OPA logs, I can't tell why it's failing.

**Comment by tsandall** [comment]
I think that in any case, we need to improve kube-mgmt to perform a full resync in case a write against OPA fails (because it's not going to retry until the resource changes.)

**Comment by tsandall** [comment]
This should be fixed by #8.
